connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala

```diff
@@ -64,6 +69,94 @@ private[client] object GrpcExceptionConverter extends JsonUtils {
     }
   }
 
+  private def enrichError(errors: Array[Error], info: ErrorInfo): Array[Error] = {
+    val sessionId = info.getMetadataOrDefault("sessionId", null)
+    val userId = info.getMetadataOrDefault("userId", null)
```

Comment:
The idea is to make the ErrorInfo self-contained for issuing the follow up FetchErrorDetails request. Otherwise, it may make the implementation on the client side a bit more complicate since each FetchErrorDetails call needs to figure out the corresponding session id and user id

Reply:
@hvanhovell Would like to know your thoughts on this